homeassistant/components/esphomelib/__init__.py

```diff
@@ -0,0 +1,338 @@
+
+    Configuration through YAML is not supported at this time.
+    """
+    return True
+
+
+def get_entry_data(hass: HomeAssistantType,
+                   key: Union[ConfigEntry, str]) -> Dict[str, Any]:
+    """Get the per-config entry data dictionary for a key."""
+    if isinstance(key, ConfigEntry):
+        key = key.entry_id
+    return hass.data[DOMAIN].setdefault(key, {})
+
+
+async def platform_async_setup_entry(hass: HomeAssistantType,
+                                     entry: ConfigEntry,
+                                     async_add_entities,
+                                     *,
+                                     component_key,
+                                     info_type,
+                                     entity_type,
+                                     state_type,
+                                     ):
+    """Set up an esphomelib platform."""
+    get_entry_data(hass, entry).setdefault(component_key, {})
+
+    @callback
+    def async_list_entities(infos: List[Any]):
+        """Update entities of this platform when entities are listed."""
+        old_entities = get_entry_data(hass, entry)[component_key]
+        new_entities = {}
+        add_entities = []
+        for info in infos:
+            if not isinstance(info, info_type):
+                # Filter out infos that don't belong to this platform.
+                continue
+
+            if info.key in old_entities:
+                # Update existing entity
+                entity = old_entities.pop(info.key)
+            else:
+                # Create new entity
+                entity = entity_type(entry.entry_id)
+                add_entities.append(entity)
+            entity.async_info_update(info)
+            new_entities[info.key] = entity
+
+        # Remove old entities
+        for entity in old_entities.values():
+            hass.async_add_job(entity.async_remove())
+
+        get_entry_data(hass, entry)[component_key] = new_entities
+        async_add_entities(add_entities)
+
+    async_dispatcher_connect(
+        hass, DISPATCHER_ON_LIST.format(entry.entry_id), async_list_entities)
+
+    @callback
+    def async_entity_state(state):
+        """Notify the appropriate entity of an updated state."""
+        entities = get_entry_data(hass, entry)[component_key]
+        if not isinstance(state, state_type):
+            return
+        if state.key not in entities:
+            return
+        entity = entities[state.key]
+        entity.async_state_update(state)
+
+    async_dispatcher_connect(
+        hass, DISPATCHER_ON_STATE.format(entry.entry_id), async_entity_state)
+
+    @callback
+    def async_device_state():
+        """Schedule a state update when the device settings change."""
+        entities = get_entry_data(hass, entry)[component_key]
+        for entity in entities.values():
+            if entity.hass is not None:
+                entity.async_schedule_update_ha_state()
```

Comment:
Send a signal here instead. In the entity connect the state scheduling callback in `async_added_to_hass`.